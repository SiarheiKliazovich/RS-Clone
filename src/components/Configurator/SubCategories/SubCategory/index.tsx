import { FunctionComponent } from "react";
import { SubCategoryType } from "../../../../types/configurator";
import { nightTheme } from "../../../../data/constants";

const SubCategory: FunctionComponent<SubCategoryType> = ({
  setType,
  subCategory,
  accentColor,
  isNavbarNightMode,
}: SubCategoryType) => {
  const backgroundColor = nightTheme.background.element;

  return (
    <li
      className="image-item"
      key={subCategory.id}
      onClick={() => setType(subCategory.type)}
    >
      <div className="img-wrapper">
        <img className="img-subcategories" src={subCategory.image} alt="img" />
      </div>
      <div className="btn-wrapper">
        <button
          style={{
            backgroundColor: isNavbarNightMode ? backgroundColor : accentColor,
          }}
          className="category-name"
        >
          {subCategory.name}
        </button>
      </div>
    </li>
  );
};
export default SubCategory;
