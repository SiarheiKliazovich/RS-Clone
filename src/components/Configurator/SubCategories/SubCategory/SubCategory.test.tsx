import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import SubCategory from ".";

describe("SubCategory", () => {
  const subCategory = {
    id: 1,
    type: "type",
    name: "name",
    image: "image.png",
  };
  const category = "category";
  const accentColor = {
    static: "#FFFFFF",
    hover: "#000000",
  };
  const isNavbarNightMode = false;

  it("renders correctly", () => {
    const { container } = render(
      <Router>
        <SubCategory
          subCategory={subCategory}
          category={category}
          accentColor={accentColor}
          isNavbarNightMode={isNavbarNightMode}
        />
      </Router>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
