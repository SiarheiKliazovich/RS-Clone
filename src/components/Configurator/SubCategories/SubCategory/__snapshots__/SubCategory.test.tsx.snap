// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubCategory renders correctly 1`] = `
<li
  class="image-item"
>
  <div
    class="img-wrapper"
  >
    <img
      alt="img"
      class="img-subcategories"
      src="image.png"
    />
  </div>
  <div
    class="btn-wrapper"
  >
    <a
      class="category-name"
      href="/configurator/category/type"
      style="background-color: rgb(255, 255, 255);"
    >
      name
    </a>
  </div>
</li>
`;
